Replace deprecated `view_*()` method from examples

rahul713rk opened the pull request:
# References and relevant issues
Closes #8080 #7047 

# Description
Replace `viewer = napari.view_image(...)` with 
```
viewer = napari.Viewer
layer = viewer.add_image(...) 
```

`.add_surfaces(...)`  for surface  
`.add_points(...)` for points

Commit:
Replace  from `view_image` to `add_image`

Commit:
Update examples to use `add_image` API instead of `view_image`

Commit:
Refactor examples to use `napari.Viewer()` and `add_points`/`add_surface` methods instead of deprecated functions

Commit:
Fix viewer initialization and update ndisplay setting in examples

Commit:
Fix variable assignment for image layer in labels3d example

Czaki commented on examples/action_manager.py:
```diff
@@ -38,7 +38,8 @@ def rotate45(viewer: napari.Viewer):
 
 # create the viewer with an image
-viewer = napari.view_image(data.astronaut(), rgb=True)
+viewer = napari.Viewer()
+layer = viewer.add_image(data.astronaut(), rgb=True)
```
Will review later. But we still keep the `imshow` method. Why not use it here?

rahul713rk replied:
https://github.com/napari/napari/issues/8080#issue-3213971533 here @TimMonko mentioned that `add_*` is preferred .

Czaki replied:
@TimMonko @DragaDoncila opinion here?

TimMonko replied:
I think that, as I mentioned in #8080, its a preference thing. In my limited experience, users without a matplotlib imshow experience, that they have a better understanding on the two line method, especially because its generic to all layer types. I personally prefer `viewer = napari.Viewer(); layer = viewer.add_*`, but I'm ok to hear other opinions.

DragaDoncila replied:
I'm with Tim on this one tbh

willingc replied:
Since these are examples, I agree with Tim and Draga on the more readable two line command.

Perhaps adding another example in the future to illustrate the point that approach `imshow` is an alternative approach to `add_*`

TimMonko commented on examples/image-points-3d.py:
```diff
@@ -14,9 +14,11 @@
+viewer = napari.Viewer()
+layer = viewer.add_image(
+        cells, channel_axis=1, name=['membranes', 'nuclei']
         )
```
```suggestion
membranes, nuclei = viewer.add_image(
    cells, channel_axis=1, name=['membranes', 'nuclei']
)
```
So, this is the great thing about `add_image` in that it returns the layers, which we didn't have before. This still passed the tests because its technically ok, but in reality, `add_image` will return a tuple of the length of the number of layers added. So in this case, we get two layers, one named membranes and one named nuclei. You could call these `layer0, layer1` or `membranes, nuclei` in this case, it doesn't matter because the notebook doesn't refer to it, but it'd be good for us to model the behavior in the example!

rahul713rk replied:
thanks @TimMonko , for mentioning this . Going to revisit the code and correct my mistakes

TimMonko commented on examples/image_border.py:
```diff
@@ -20,10 +20,9 @@
-)
+# Create a napari viewer with a 3D, 2 channel image, with the `add_image` convenience function.
+viewer = napari.Viewer()
+layer = viewer.add_image(data.cells3d(), channel_axis=1, name=['membrane', 'nuclei'])
```
```suggestion
membrane, nuclei = viewer.add_image(data.cells3d(), channel_axis=1, name=['membrane', 'nuclei'])
```
The best part about this, is then you can then update the rest of the example to use
`membrane.` and `nuclei.` For example, instead of `viewer.layers[0].` -- a great way to update the example for napari best practices! 
Let me know if you need more guidance :)

rahul713rk committed:
Update examples/image-points-3d.py

Co-authored-by: Tim Monko <timmonko@gmail.com>

rahul713rk committed:
Update examples/image_border.py

Co-authored-by: Tim Monko <timmonko@gmail.com>

TimMonko commented on examples/mpl_plot_.py:
```diff
@@ -17,7 +17,8 @@
-viewer = napari.view_image(img, colormap='viridis')
+viewer = napari.Viewer()
+layer = viewer.add_image(img, colormap='viridis')
 layer = viewer.layers[-1]
```
this is the strength of the new way! we can remove this line now :) 
```suggestion
```

TimMonko reviewed (commented):
I didn't totally finish reviewing, but check generally over my comments to see if it applies to other files! Great work so far! 
Namely, check that `channel_axis` returns objects with the length of layers.
And, now that we return the layer, we use its reference rather than the the indexing version, so for example the returned `image` rather than `viewer.layers[idx]`. It would replace lots more code but show a great thing that took me a while to learn!

rahul713rk committed:
Update examples/mpl_plot_.py

Co-authored-by: Tim Monko <timmonko@gmail.com>

willingc commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,13 @@
 astronaut = data.astronaut()
-# Also view_path, view_shapes, view_points, view_labels etc.
-viewer_b = napari.view_image(astronaut, name='astronaut')
+# Also view_path, add_shapes, add_points, add_labels etc.
```
In this example a second viewer needs to be created so that there are two viewer windows `viewer_a` and `viewer_b. I believe the example's purpose is to demonstrate that multiple viewers are possible.

psobolewskiPhD replied:
exactly, so yes need add:
```
viewer_b = napari.Viewer()
```

willingc reviewed (commented):
Overall, I think this PR makes sense. Please take a look again at the multiple viewer example since I think that we need to instantiate two distinct viewers. Thanks!

willingc commented on examples/mpl_plot_.py:
```diff
@@ -19,7 +19,6 @@
 # add it to the viewer
 viewer = napari.Viewer()
 layer = viewer.add_image(img, colormap='viridis')
-layer = viewer.layers[-1]
```
Just curious, why is this line being removed?

rahul713rk replied:
because `viewer.add_image` returns `Layers` which i already assigned in L21  , the below code is redundant and do nothing.

willingc replied:
Thanks @rahul713rk

psobolewskiPhD commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,13 @@
 # add the image
 photographer = data.camera()
-viewer_a = napari.view_image(photographer, name='photographer')
+viewer = napari.Viewer()
```
```suggestion
viewer_a = napari.Viewer()
```
This needs to be viewer_a and the other one, as noted below by Carol, needs to be viewer_b

psobolewskiPhD commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,13 @@
 photographer = data.camera()
-viewer_a = napari.view_image(photographer, name='photographer')
+viewer = napari.Viewer()
+layer_a = viewer.add_image(photographer, name='photographer')
```
```suggestion
layer_a = viewer_a.add_image(photographer, name='photographer')
```

psobolewskiPhD commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,13 @@
-# Also view_path, view_shapes, view_points, view_labels etc.
-viewer_b = napari.view_image(astronaut, name='astronaut')
+# Also view_path, add_shapes, add_points, add_labels etc.
+layer_b = viewer.add_image(astronaut, name='astronaut')
```
```suggestion
layer_b = viewer_b.add_image(astronaut, name='astronaut')
```

Czaki committed:
Improve performance and memory usage of editing Shapes layer  (#8006)

# References and relevant issues

Pointed
https://github.com/napari/napari/pull/7995#issuecomment-2955694555

# Description

Instead of storing information about which vertex/triangle belongs to a
given shape, store information about which range of vertices/triangles
belongs to a given shape.

When editing a shape, try to do as much updating in place as possible
instead of splitting and concatenating the array.

Update typing in `shape_list.py`

A benchmark showing that the used solution is effective. 

Use `batched_updates` context manager to reduce call of
`_update_displayed`

---------

Co-authored-by: Carol Willing <carolcode@willingconsulting.com>

willingc committed:
Revert #8006 Improve performance and memory usage of editing Shapes layer (#8104)

This reverts commit 04009aa30bc01f47ccd673fa15c2da6b252d618f.

SciPy sprint found that the behavior had unexpected results.

rahul713rk committed:
Update examples/multiple_viewers_.py

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

rahul713rk committed:
Update examples/multiple_viewers_.py

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

rahul713rk committed:
Update examples/multiple_viewers_.py

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

Commit:
Fix viewer initialization in multiple viewers example

Commit:
Fix variable assignment for image layers in labels3d example

brisvag commented on examples/image-points-3d.py:
```diff
@@ -14,9 +14,11 @@
+membranes, nuclei = viewer.add_image(
+    cells, channel_axis=1, name=['membranes', 'nuclei']
+)
+viewer.dims.ndisplay = 3
```
Alternatively:

```suggestion
viewer = napari.Viewer(ndisplay=3)
membranes, nuclei = viewer.add_image(
    cells, channel_axis=1, name=['membranes', 'nuclei']
)
```

brisvag commented on examples/nD_shapes_with_text.py:
```diff
@@ -12,7 +12,9 @@
-viewer = napari.view_image(blobs.astype(float), ndisplay=3)
+viewer = napari.Viewer()
+layer = viewer.add_image(blobs.astype(float))
+viewer.dims.ndisplay = 3
```
```suggestion
viewer = napari.Viewer(ndisplay=3)
layer = viewer.add_image(blobs.astype(float))
```

rahul713rk replied:
thanks @brisvag

brisvag reviewed (approved):
> Apologies for the Git confusion — I tried --rebase on the remote for the first time and ran into some issues.

No worries, no one survives unscathed their first rebase :P

rahul713rk committed:
Update examples/image-points-3d.py

Co-authored-by: Lorenzo Gaifas <brisvag@gmail.com>

rahul713rk committed:
Update examples/nD_shapes_with_text.py

Co-authored-by: Lorenzo Gaifas <brisvag@gmail.com>

psobolewskiPhD commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,14 @@
 # add the image
 photographer = data.camera()
-viewer_a = napari.view_image(photographer, name='photographer')
+viewer_a = napari.Viewer()
```
```suggestion
viewer_a = napari.Viewer(title="viewer a")
```
Totally optional, just noticed we don't do this.

psobolewskiPhD commented on examples/multiple_viewers_.py:
```diff
@@ -13,12 +13,14 @@
-# Also view_path, view_shapes, view_points, view_labels etc.
-viewer_b = napari.view_image(astronaut, name='astronaut')
+# Also view_path, add_shapes, add_points, add_labels etc.
+viewer_b = napari.Viewer()
```
```suggestion
viewer_b = napari.Viewer(title="viewer b")
```
same as above

psobolewskiPhD reviewed (approved):
Thanks!
I made a tiny suggestion, but it's technically out scope of this PR. Still makes life easier with the multiple viewer example, so maybe worth it.

rahul713rk committed:
Update examples/multiple_viewers_.py

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

rahul713rk committed:
Update examples/multiple_viewers_.py

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

TimMonko reviewed (approved):
Another great and helpful PR @rahul713rk ! Thank you thank you! You're really cruising lately.
